### PR TITLE
fix: updated cdx purl parsing to match spec

### LINKED ISSUE #354: unable to assemble graphs: Node ... has no value for property purl
I'm having an issue with importing a cyclone DX SBOM.

```
{"level":"error","ts":1674627147.1044576,"caller":"collector/collector.go:87","msg":"emit error: unable to assemble graphs: Node {ianmlewis/slsa-and-guac-demo:0.0.1 []   [] [] {file:///docs/cyclonedx/syft-cyclonedx-docker.io-ianmlewis-slsa-and-guac-demo:0.0.1.json FileCollector}} has no value for property purl","stacktrace":"github.com/guacsec/guac/pkg/handler/collector.Collect\n\t/usr/local/google/home/ianlewis/go/src/github.com/guacsec/guac/pkg/handler/collector/collector.go:87\ngithub.com/guacsec/guac/cmd/guacone/cmd.glob..func5\n\t/usr/local/google/home/ianlewis/go/src/github.com/guacsec/guac/cmd/guacone/cmd/files.go:168\ngithub.com/spf13/cobra.(*Command).execute\n\t/usr/local/google/home/ianlewis/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/usr/local/google/home/ianlewis/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044\ngithub.com/spf13/cobra.(*Command).Execute\n\t/usr/local/google/home/ianlewis/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968\ngithub.com/guacsec/guac/cmd/guacone/cmd.Execute\n\t/usr/local/google/home/ianlewis/go/src/github.com/guacsec/guac/cmd/guacone/cmd/root.go:104\nmain.main\n\t/usr/local/google/home/ianlewis/go/src/github.com/guacsec/guac/cmd/guacone/main.go:23\nruntime.main\n\t/usr/local/google/home/ianlewis/opt/go/src/runtime/proc.go:250"}
```

The SBOMs I'm trying to import are here
https://github.com/ianlewis/slsa-and-guac-demo/tree/main/docs/cyclonedx

I couldn't find a material difference from the demo SBOMs in [guac-data](https://github.com/guacsec/guac-data/tree/main/docs/cyclonedx). Any idea what's going on here?

### PR DESCRIPTION
**Pull request by pxp928**
Signed-off-by: pxp928 <parth.psu@gmail.com>

Updated CDX heuristic to handle `docker.io` use case and match the [oci purl type](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#oci). 

Updated `setup.md` with the changes to the CDX parser.

fixes #354 
fixes https://github.com/guacsec/guac/issues/360

### CODE REVIEW

File: pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go [hunk 1]
```diff
@@ -113,10 +113,25 @@ func (c *cyclonedxParser) addRootPackage(cdxBom *cdx.BOM) {
 		} else {
 			splitImage := strings.Split(cdxBom.Metadata.Component.Name, "/")
 			if len(splitImage) == 3 {
-				rootPackage.Purl = "pkg:oci/" + splitImage[2] + "?repository_url=" + splitImage[0] + "/" + splitImage[1]
-				rootPackage.Version = cdxBom.Metadata.Component.Version
-				rootPackage.Digest = append(rootPackage.Digest, cdxBom.Metadata.Component.Version)
-				rootPackage.Tags = []string{"CONTAINER"}
+				// example: gcr.io/distroless/static:nonroot
+				splitTag := strings.Split(splitImage[2], ":")
```
Comment (lumjjb) [inline pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go:117]: may need to check if this exists, if tag doesn't exist then may not need to include. can include 1 test with no tag as well

**Comment by pxp928** [comment]
Updated setup.md with the changes to the CDX parser.

**Review by lumjjb — APPROVED** [review]
lgtm

**Review by mlieberman85 — APPROVED** [review]
lgtm

**Pull request merged**